xvec/accessor.py

```diff
@@ -918,6 +918,375 @@ def to_geodataframe(
+            import rasterio
+        except ImportError as err:
+            raise ImportError(
+                "The rasterio package is required for `xvec.agg_geom()`. "
+                "You can install it using 'conda install -c conda-forge rasterio' or "
+                "'pip install rasterio'."
+            ) from err
+
+        try:
+            import gc
+        except ImportError as err:
+            raise ImportError(
+                "The gc package is required for `xvec.agg_geom()`. "
+                "Make sure 'gc' is included in the standard library"
+                "Check your Python installation for any issues."
+            ) from err
+
+        # Create a GeoSeries from the geometry
+        geo_series = gpd.GeoSeries(geom)
+
+        # Convert the GeoSeries to a GeometryArray
+        geometry_array = geo_series.geometry.array
+
+        xar_chunk = self._obj[var]
+        mask = rasterio.features.geometry_mask(
+            geometry_array,
+            out_shape=(xar_chunk.shape[0], xar_chunk.shape[1]),
+            transform=trans,
+        )
+
+        masked_data = xar_chunk * mask[:, :, np.newaxis]
+        del mask, xar_chunk
+        gc.collect()
+
+        if dask:
+            try:
+                import dask.array as da
+            except ImportError as err:
+                raise ImportError(
+                    "The dask package is required for This step. "
+                    "You can install it using 'conda install -c conda-forge dask' or "
+                    "'pip install dask'."
+                ) from err
+
+            if stat == "sum":
+                stat_within_polygons = da.sum(masked_data, axis=(x_axis, y_axis))
+            elif stat == "mean":
+                stat_within_polygons = da.mean(masked_data, axis=(x_axis, y_axis))
+            elif stat == "median":
+                stat_within_polygons = da.median(masked_data, axis=(x_axis, y_axis))
+            elif stat == "max":
+                stat_within_polygons = da.max(masked_data, axis=(x_axis, y_axis))
+            elif stat == "min":
+                stat_within_polygons = da.min(masked_data, axis=(x_axis, y_axis))
+
+            result = stat_within_polygons.compute()
+
+        else:
+            if stat == "sum":
+                stat_within_polygons = masked_data.sum(dim=[y_coords, x_coords])
+            elif stat == "mean":
+                stat_within_polygons = masked_data.mean(dim=[y_coords, x_coords])
+            elif stat == "median":
+                stat_within_polygons = masked_data.median(dim=[y_coords, x_coords])
+            elif stat == "max":
+                stat_within_polygons = masked_data.max(dim=[y_coords, x_coords])
+            elif stat == "min":
+                stat_within_polygons = masked_data.min(dim=[y_coords, x_coords])
+
+            result = stat_within_polygons.values
+
+        del masked_data, stat_within_polygons
+        gc.collect()
+
+        return result
+
+    def spatial_agg(
```

Comment:
```suggestion
    def _spatial_agg(
```
This also should be private.